pkgs/tools/video/recyclarr/default.nix

```diff
@@ -0,0 +1,93 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, autoPatchelfHook
+, fixDarwinDylibNames
+, darwin
+, recyclarr
+, git
+, icu
+, testers
+, zlib
+,
+}:
+let
+  os =
+    if stdenv.isDarwin
+    then "osx"
+    else "linux";
+  arch =
+    {
+      x86_64-linux = "x64";
+      aarch64-linux = "arm64";
+      x86_64-darwin = "x64";
+      aarch64-darwin = "arm64";
+    }."${stdenv.hostPlatform.system}"
+      or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+
+  hash =
+    {
+      x64-linux_hash = "sha256-aEcinTJlO++rTeyqGJea0TWtleH6fyooA8RhT0Qj24c=";
+      arm64-linux_hash = "sha256-9Gyk09zAGzahP8FCGxj037vaK8h++3M5R2Qqop99Gs4=";
+      x64-osx_hash = "sha256-c87eOZBz+RtbIi+dlXKKVMyPI8JqYDuiaL4xOkDRFn0=";
+      arm64-osx_hash = "sha256-zSHgLXRDB6UA7V0LFgLq9ChqB40IHIJJxRqAYyVFlB8=";
+    }."${arch}-${os}_hash";
+
+  libPath = {
+    osx = "DYLD_LIBRARY_PATH : ${lib.makeLibraryPath [darwin.ICU zlib]}";
+    linux = "LD_LIBRARY_PATH : ${lib.makeLibraryPath [icu zlib]}";
+  }."${os}";
+
+in
+stdenv.mkDerivation rec {
+  pname = "recyclarr";
+  version = "5.1.0";
+
+  src = fetchurl {
+    url = "https://github.com/recyclarr/recyclarr/releases/download/v${version}/recyclarr-${os}-${arch}.tar.xz";
+    inherit hash;
+  };
```

Comment:
Please use the `fetchFromGitHub` wrapper method https://nixos.org/manual/nixpkgs/stable/#fetchfromgithub

Reply:
Is there a way to use `fetchFromGitHub` to get an archive from the Releases page? It's my understanding that `fetchFromGitHub` pulls a copy of the source code associated w/ a particular tag. What I'm aiming for here is downloading the compiled binaries from the Releases page (generated by the recyclarr repo CI). 

Right now, I'm using `fetchurl` as compiling from source 1) adds a significant amount of CPU/time/storage demands to download entire dotnet SDK for multiple platforms, and 2) requires patching several portions of source code related to the project trying to parse its own git history to generate a version string (these checks must be patched out since `fetchFromGitHub` does not pull the entire project's history).

Reply:
Usually I would say building from source is the prefered way even if it is more compute intense, that's why we have cache.nixos.org and the part with it trying to read stuff from it's git history get's patched in many packages but I'm to deep out of my comfort zone to make a clear cut decision so I would rather have some one else review this again. Hope that's fine with you?